src/database/schemas.ts

```diff
@@ -220,11 +220,21 @@ export default class Schemas {
    * @param object Not user input
    */
   static async generateSQL(schema: string, object: string, internalType: string): Promise<string> {
-    const lines = await JobManager.runSQL<{ SRCDTA: string }>([
-      `CALL QSYS2.GENERATE_SQL(?, ?, ?, CREATE_OR_REPLACE_OPTION => '1', PRIVILEGES_OPTION => '0')`
+    await JobManager.runSQL<{ SRCDTA: string }>([
+      `CALL QSYS2.GENERATE_SQL( DATABASE_OBJECT_NAME => ?, DATABASE_OBJECT_LIBRARY_NAME => ?, DATABASE_OBJECT_TYPE => ?
+                              , CREATE_OR_REPLACE_OPTION => '1', PRIVILEGES_OPTION => '0'
+                              , DATABASE_SOURCE_FILE_NAME => '*STMF'
+                              , STATEMENT_FORMATTING_OPTION => '0'
+                              , SOURCE_STREAM_FILE => '/tmp/Q_GENSQL_' concat current_user concat '.sql'
+                              , SOURCE_STREAM_FILE_END_OF_LINE => 'LF'
+                              , SOURCE_STREAM_FILE_CCSID => 1208 )`
     ].join(` `), { parameters: [object, schema, internalType] });
+    const lines = await JobManager.runSQL<{ LINE: string }>(
+      `select LINE
+         from table( QSYS2.IFS_READ( PATH_NAME => '/tmp/Q_GENSQL_' concat current_user concat '.sql' ) )`
```

Comment:
At this point, should we just use our own Code for IBM i API to get the contents of the file?

Reply:
We could... and we could also use our own C4i API to run the SQL and get the result in one step instead of these multiple calls. And maybe use the temporary streamfile API?

I'm quite busy currently with bigger projects at work, so we either 
1. wait for me to get some spare time
2. let you code it or
3. put a TODO in the source.

Which do you prefer? 😉

Reply:
I am happy to do it actually! I will have a go at it on Monday morning. Using the temp streamfile is also a good idea.